unitex/src/fr/umlv/unitex/utils/SingleInstanceMonitor.java

```diff
@@ -0,0 +1,248 @@
+package fr.umlv.unitex.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintWriter;
+import java.io.Writer;
+import java.net.BindException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.net.ServerSocket;
+import java.net.Socket;
+import java.net.SocketAddress;
+import java.net.SocketException;
+import java.util.ArrayList;
+
+import javax.swing.JFrame;
+import javax.swing.SwingWorker;
+
+import fr.umlv.unitex.Unitex;
+import fr.umlv.unitex.config.Config;
+import fr.umlv.unitex.frames.UnitexFrame;
+
+
+/**
+ * This class ensures only one instance of the IDE can run at one time
+ * @author markpower
+ *
+ */
+public class SingleInstanceMonitor {
+
+	private ServerSocket serverSocket = null;
+	private Socket socket = null;
+	private final String DELIMITER = ";";
+	private boolean SOCKET_OPEN = false;
+	private final String PORT_FILE_NAME = "single_instance.port";
```

Comment:
If the user does not have write access to the directory where Unitex.jar or Gramlab.jar is located, try to creating "single_instance.port" will raise an exception. Under Windows this is a common situation when Unitex/GramLab is installed for multiple users on the same computer. Under Linux, you could simulate this situation doing for example:

 ```
mkdir /tmp/test
mv Unitex.jar Gramlab.jar /tmp/test
cd /tmp/test
sudo chown root:root /tmp/test
java -jar Unitex.jar
```

The output:
```
Error writing to port file for single instance listener running on port 34693
java.io.FileNotFoundException: /tmp/test/single_instance.port (Permission denied)
	at java.io.FileOutputStream.open0(Native Method)
	at java.io.FileOutputStream.open(FileOutputStream.java:270)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:213)
	at java.io.FileOutputStream.<init>(FileOutputStream.java:162)
	at java.io.FileWriter.<init>(FileWriter.java:90)
	at fr.umlv.unitex.utils.SingleInstanceMonitor.writePortFile(Unknown Source)
	at fr.umlv.unitex.utils.SingleInstanceMonitor.isRunning(Unknown Source)
	at fr.umlv.unitex.Unitex.main(Unknown Source)
```

To avoid this situation use instead .unitex.cfg and .gramlab appending the next settings:

```
allow_only_one_application_instance = true
one_application_instance_port_number = 49627
```